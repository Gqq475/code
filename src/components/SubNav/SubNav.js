import React from 'react'
import classes from './SubNav.scss'

type Props = {
  params: Object,
  location: String
}
export class SubNav extends React.Component {
  props: Props;

  render () {
    return (
      <div className={classes['SubNav-container']}>
        <div>
          <ul className={`pt-breadcrumbs ${classes.breadcrumbs}`}>
            <li><span className={`pt-breadcrumb ${classes.breadcrumbs}`}>{_.HOME}</span></li>
            <li><span className={`pt-breadcrumb ${classes.breadcrumbs}`}>{this.props.location}</span></li>
            <li>
              <span className={`pt-breadcrumb pt-breadcrumb-current ${classes.currentPage}`}>
                {{
                  currentalarm: _.CURRENTALARM,
                  alarmhistory: _.ALARMHISTORY,
                  criteriamanager: _.CRITERIAMANAGER,
                  defectjudge: _.DEFECTJUDGE,
                  base: _.通用功能,
                  defectmap: _.DEFECTMAP,
                  stk: _.搬运路径共通性分析,
                  dashboard: _.DASHBOARD,
                  alarmrate: _.ALARMRATE,
                  defectrate: _.DEFECTRATE分析,
                  echarts: 'echarts制图试用',
                  autodefectjudge: _.AUTODEFECTJUDGE,
                  servicemanagement: _.SERVICEMANAGEMENT,
                  cpk: _.制程能力推移图,
                  spc: '平均值-全距管制图',
                  labelingtool: '标记工具',
                  modelversion: '模型管理',
                  modeltest: '模型检测',
                  modeltrain: '模型训练'
                }[this.props.params.path]}
              </span>
            </li>
          </ul>
        </div>
      </div>
    )
  }
}

export default SubNav
