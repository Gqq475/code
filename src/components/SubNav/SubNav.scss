.SubNav-container {
  height: 50px;
  background: #fff;
  border-bottom: solid 2px #0073BB;
  box-shadow: inset 0 -20px 40px #d7d7d7;
  padding: 0 80px;
  display: flex;
  align-items: center;


  .breadcrumbs {
    line-height: 20px;
    font-size: 13px;
    color: #727272;
  }

  .currentPage {
    color: #0073BB;
  }
}
