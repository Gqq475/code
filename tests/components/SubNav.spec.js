import React from 'react'
import { SubNav } from 'components/SubNav/SubNav'

describe('(Component) SubNav', () => {
  it('render normally', () => {
    const props = {
      'location': 'SADJ',
      'params': {
        'path': 'autodefectjudge'
      }
    }
    const wrapper = shallow(
      <SubNav {...props} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
