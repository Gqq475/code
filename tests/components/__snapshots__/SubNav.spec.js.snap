// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) SubNav render normally 1`] = `
<div
  className="SubNav-container"
>
  <div>
    <ul
      className="pt-breadcrumbs breadcrumbs"
    >
      <li>
        <span
          className="pt-breadcrumb breadcrumbs"
        />
      </li>
      <li>
        <span
          className="pt-breadcrumb breadcrumbs"
        >
          SADJ
        </span>
      </li>
      <li>
        <span
          className="pt-breadcrumb pt-breadcrumb-current currentPage"
        />
      </li>
    </ul>
  </div>
</div>
`;
